tests/eas/st_hold.py

```diff
@@ -0,0 +1,408 @@
+#    Copyright 2018 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import logging
+import unittest
+
+from env import TestEnv
+from trappy import FTrace
+from wlgen import Periodic, RTA
+from bart.common.Utils import area_under_curve
+
+test_conf = { "modules": ["bl", "cpufreq", "cgroups" ],
+              "tools": ["rt-app"],
+              "ftrace" : {
+                  "events" : ["cpu_frequency", "sched_switch", "sched_boost_cpu"],
+                  "buffsize" : 10240
+              },
+              "flags" : [ "ftrace" ]
+            }
+
+BOOSTED_CGROUP_NAME = '/boosted'
+UNBOOSTED_CGROUP_NAME = '/unboosted'
+FREQ_MARGIN = 20000
+TEST_CPU = 0
+
+class _SchedtuneHoldTest(unittest.TestCase):
+    """
+    Base class for all Schedtune-Hold feature tests. Contains
+    setup code and useful test methods.
+    When a derived class is instantiated, this sequence of events
+    happens:
+      _SchedtuneHoldTest:setUpClass called
+          -> connect target object
+          -> call derived_class:child_check_config() if implemented
+          -> call derived_class:populate_params() (must be implemented)
+          -> run workload
+          -> build Trace object in self.trace_obj
+    Then the actual test function is executed on the collected trace.
+
+    child_check_config is used for the two tests present where only one
+    can be run - which one is OK depends upon the status of the
+    SCHEDTUNE_BOOST_HOLD_ALL sched_feature. This function should raise a
+    unittest.SkipTest exception if it wants to prevent a derived class test
```

Comment:
I think for external stuff  ``:class:`unittest.SkipTest` `` should work. I don't think it's as important to have proper links for those though - people usually already know them. It's more about making it easier for people new to LISA's codebase.

Reply:
ok